### PR TITLE
Update skipper version, step 1/2

### PR DESCRIPTION
**Pull request by RomanZavodskikh**
+ routesrv: require kubernetes option 
+ doc: admissionControl response status code 
+ filters/builtin: performance optimization for sanitize in StripQuery …
+ cmd/skipper: fix version and commit for go install
+ Passive Health Check is implemented, additional config item for this feature is provided

FYI https://github.com/zalando/skipper/compare/v0.20.15...v0.21.4

### CODE REVIEW

**Comment by RomanZavodskikh** [comment]
:+1:

File: cluster/manifests/skipper/deployment.yaml [hunk 1]
```diff
@@ -301,6 +301,9 @@ spec:
 {{ if .Cluster.ConfigItems.skipper_ingress_inline_routes }}
           - "-inline-routes={{ .Cluster.ConfigItems.skipper_ingress_inline_routes }}"
 {{ end }}
+{{ if .Cluster.ConfigItems.skipper_ingress_passive_health_check_options }}
```
Comment (szuecs) [inline cluster/manifests/skipper/deployment.yaml]: I don't think that's correct

File: cluster/config-defaults.yaml [hunk 2]
```diff
@@ -70,6 +70,7 @@ skipper_ingress_max_replicas: "50"
 {{end}}
 skipper_ingress_cpu: "1000m"
 skipper_ingress_memory: "1500Mi"
+skipper_ingress_passive_health_check_options: ""
```
Comment (szuecs) [inline cluster/config-defaults.yaml]: Let's have test clusters with a working configuration

**Review by szuecs — CHANGES_REQUESTED** [review]

File: cluster/manifests/skipper/deployment.yaml (hunk 1 above)
Comment (RomanZavodskikh) [reply cluster/manifests/skipper/deployment.yaml]: I don't know which problem you had in your mind, but I see the following, imagine:
+ we merge this PR (only canary)
+ I set `skipper_ingress_passive_health_check_options` config item to something like `period=1s,min-requests=10,max-drop-probability=0.9`
+ CLM sets this option both to canary and main fleet
+ main fleet version is older that version where PHC was introduced -> main fleet cannot pars cmdline option -> main fleet doesn't roll out

**Comment by RomanZavodskikh** [comment]
@szuecs I have tested the latest commit (`ecebdde0f`) on one of our test clusters, the skipper pod with newest version in canary works with cmdline args without any issues.
```
$ zkubectl -n kube-system get pod -lapplication=skipper-ingress -lcomponent=ingress
NAME                                      READY   STATUS    RESTARTS   AGE
skipper-ingress-6f47f9d6c-49xtq           1/1     Running   0          3h30m
skipper-ingress-canary-5c678bcc4b-zr9kj   1/1     Running   0          33m
$ zkubectl -n kube-system get pod skipper-ingress-canary-5c678bcc4b-zr9kj -o yaml | grep 'passive'
    - -passive-health-check=period=10s,min-requests=10,max-drop-probability=0.9
$
```

File: cluster/config-defaults.yaml (hunk 2 above)
Comment (RomanZavodskikh) [reply cluster/config-defaults.yaml]: https://github.com/zalando-incubator/kubernetes-on-aws/pull/7063#issuecomment-1963824510

File: cluster/manifests/skipper/deployment.yaml (hunk 1 above)
Comment (szuecs) [reply cluster/manifests/skipper/deployment.yaml]: If you set it to `""` it's likely also `true`, not sure but I think we do not use the implicit empty string test. I think we use `index .foo > 0` or something like that. Please check it.

**Comment by szuecs** [comment]
> @szuecs I have tested the latest commit (`ecebdde0f`) on one of our test clusters, the skipper pod with newest version in canary works with cmdline args without any issues.
> 
> ```
> $ zkubectl -n kube-system get pod -lapplication=skipper-ingress -lcomponent=ingress
> NAME                                      READY   STATUS    RESTARTS   AGE
> skipper-ingress-6f47f9d6c-49xtq           1/1     Running   0          3h30m
> skipper-ingress-canary-5c678bcc4b-zr9kj   1/1     Running   0          33m
> $ zkubectl -n kube-system get pod skipper-ingress-canary-5c678bcc4b-zr9kj -o yaml | grep 'passive'
>     - -passive-health-check=period=10s,min-requests=10,max-drop-probability=0.9
> $
> ```

Can you also test what will happen if you set the config item to empty string?

**Comment by RomanZavodskikh** [comment]
> Can you also test what will happen if you set the config item to empty string?

It works and creates skipper-ingress canary pod with PHC disabled.
```
$ zkubectl -n kube-system get pod skipper-ingress-canary-77596d7944-tjc7x 
NAME                                      READY   STATUS    RESTARTS   AGE
skipper-ingress-canary-77596d7944-tjc7x   1/1     Running   0          4m29s
$ zkubectl -n kube-system get pod skipper-ingress-canary-77596d7944-tjc7x -o yaml | grep 'passive'
$
```

File: cluster/manifests/skipper/deployment.yaml (hunk 1 above)
Comment (RomanZavodskikh) [reply cluster/manifests/skipper/deployment.yaml]: > we do not use the implicit empty string test

We literally do this three lines above this line:
```
{{ if .Cluster.ConfigItems.skipper_ingress_inline_routes }}
          - "-inline-routes={{ .Cluster.ConfigItems.skipper_ingress_inline_routes }}"
{{ end }}
```

As far as I see, inline routes work for us for years without problem, so I expect the option provided in this PR to work the same way.

**Comment by RomanZavodskikh** [comment]
> Can you also test what will happen if you set the config item to empty string?

I have tested four possible options which seem to cover all possible scenarios. All four options make CLM deploy the exactly what is expected:
+ if there is some valid PHC configuration, the main fleet has no PHC configured, the canary has PHC configured
+ if there is empty line set, the main fleet has no PHC configured, the canary has no PHC configured as well
+ if there is no key, the main fleet has no PHC configured, the canary has no PHC configured as well
+ if there is some invalid PHC configuration, the main fleet has no PHC configured, the canary does not start

```
# 'period=10s,min-requests=10,max-drop-probability=0.9'
$ kubectl -n kube-system get pod -lapplication=skipper-ingress -lcomponent=ingress
NAME                                      READY   STATUS    RESTARTS   AGE
skipper-ingress-6f47f9d6c-49xtq           1/1     Running   0          5h35m
skipper-ingress-canary-5c678bcc4b-lf7bk   1/1     Running   0          3m38s
$ kubectl -n kube-system get pod skipper-ingress-6f47f9d6c-49xtq -o yaml | grep 'passive'
$ kubectl -n kube-system get pod skipper-ingress-canary-5c678bcc4b-lf7bk -o yaml | grep 'passive'
    - -passive-health-check=period=10s,min-requests=10,max-drop-probability=0.9
$


# empty line set ''
$ kubectl -n kube-system get pod -lapplication=skipper-ingress -lcomponent=ingress
NAME                                      READY   STATUS    RESTARTS   AGE
skipper-ingress-6f47f9d6c-49xtq           1/1     Running   0          5h38m
skipper-ingress-canary-77596d7944-f4vdh   1/1     Running   0          83s
$ kubectl -n kube-system get pod skipper-ingress-6f47f9d6c-49xtq -o yaml | grep 'passive'
$ kubectl -n kube-system get pod skipper-ingress-canary-77596d7944-f4vdh -o yaml | grep 'passive'
$


# config item deleted
$ kubectl -n kube-system get pod -lapplication=skipper-ingress -lcomponent=ingress
NAME                                      READY   STATUS    RESTARTS   AGE
skipper-ingress-6f47f9d6c-49xtq           1/1     Running   0          5h40m
skipper-ingress-canary-77596d7944-nsmw9   1/1     Running   0          62s
$ kubectl -n kube-system get pod skipper-ingress-6f47f9d6c-49xtq -o yaml | grep 'passive'
$ kubectl -n kube-system get pod skipper-ingress-canary-77596d7944-nsmw9 -o yaml | grep 'passive'
$


# 'someInvalidValue'
$ kubectl -n kube-system get pod -lapplication=skipper-ingress -lcomponent=ingress
NAME                                      READY   STATUS             RESTARTS        AGE
skipper-ingress-6f47f9d6c-49xtq           1/1     Running            0               5h57m
skipper-ingress-canary-6f8554b59c-chx6v   0/1     CrashLoopBackOff   7 (2m11s ago)   15m
skipper-ingress-canary-77596d7944-nsmw9   1/1     Running            0               18m
$
```

**Review by szuecs — APPROVED** [review]

File: cluster/manifests/skipper/deployment.yaml (hunk 1 above)
Comment (szuecs) [reply cluster/manifests/skipper/deployment.yaml]: thanks

File: cluster/config-defaults.yaml (hunk 2 above)
Comment (szuecs) [reply cluster/config-defaults.yaml]: Let's have something like `if production -> empty string else "period=10s,min-requests=10,max-drop-probability=0.9"`

**Comment by RomanZavodskikh** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:
